extensions/resource-deployment/src/interfaces.ts

```diff
@@ -285,7 +285,7 @@ export interface IOptionsSource {
 export interface OptionsInfo {
 	values?: string[] | azdata.CategoryValue[],
 	source?: IOptionsSource,
-	defaultValue: string,
+	defaultValue: string | ResourceTypeOptionValue,
```

Comment:
Please make sure the [docs](https://github.com/microsoft/azuredatastudio/blob/main/extensions/resource-deployment/DEVELOPER_GUIDE.md?plain=1#L218) are updated with this new type

(while you're there the formatting seems to be off, so if you could fix that so each property is on a separate line that'd be great)